### PR TITLE
Make entrypoints configurable via elm-tooling.json

### PR DESCRIPTION
**Pull request by Razzeee**
Closes https://github.com/elm-tooling/elm-language-client-vscode/issues/114 and https://github.com/elm-tooling/elm-language-server/issues/118

Thanks to @lydell for the definition of `elm-tooling.json` https://github.com/lydell/elm-tooling.json

### CODE REVIEW

**Comment by Razzeee** [comment]
We need to mention this in the docs, so that it's discoverable.

**Comment by lydell** [comment]
I added more validation and logging: https://github.com/elm-tooling/elm-language-server/pull/316

File: src/providers/diagnostics/elmMakeDiagnostics.ts [hunk 1]
```diff
@@ -666,7 +686,7 @@ export class ElmMakeDiagnostics {
                 // elm-test might supply absolute paths to files
                 file: errorObject.path
                   ? path.relative(cwd, errorObject.path)
-                  : relativePathToFile,
+                  : relativePathToFile[0],
```
Comment (lydell) [inline src/providers/diagnostics/elmMakeDiagnostics.ts:689]: Is it OK to only use the first one here? (I have no idea what the code is doing here, sorry)

Comment (Razzeee) [reply src/providers/diagnostics/elmMakeDiagnostics.ts:689]: I think this should never really happen and as long as the fallback in, we should be safe

**Pull request merged**